apps/dashboard/app/(app)/logs/components/log-details/index.tsx

```diff
@@ -0,0 +1,100 @@
+"use client";
+import { Card, CardContent } from "@/components/ui/card";
+import { memo, useMemo, useState } from "react";
+import { useDebounceCallback } from "usehooks-ts";
+import { DEFAULT_DRAGGABLE_WIDTH } from "../../constants";
+import type { Log } from "../../types";
+import { LogFooter } from "./components/log-footer";
+import { LogHeader } from "./components/log-header";
+import ResizablePanel from "./resizable-panel";
+
+type Props = {
+  log: Log | null;
+  onClose: () => void;
+  distanceToTop: number;
+};
+
+const PANEL_WIDTH_SET_DELAY = 150;
+
+const _LogDetails = ({ log, onClose, distanceToTop }: Props) => {
+  const [panelWidth, setPanelWidth] = useState(DEFAULT_DRAGGABLE_WIDTH);
+
+  const debouncedSetPanelWidth = useDebounceCallback((newWidth) => {
+    setPanelWidth(newWidth);
+  }, PANEL_WIDTH_SET_DELAY);
+
+  const panelStyle = useMemo(
+    () => ({
+      top: `${distanceToTop}px`,
+      width: `${panelWidth}px`,
+      height: `calc(100vh - ${distanceToTop}px)`,
+      paddingBottom: "1rem",
+    }),
+    [distanceToTop, panelWidth],
+  );
+
+  if (!log) {
+    return null;
+  }
+
+  return (
+    <ResizablePanel
+      onResize={debouncedSetPanelWidth}
+      onClose={onClose}
+      className="absolute right-0 bg-background border-l border-t border-solid font-mono border-border shadow-md overflow-y-auto z-[3]"
+      style={panelStyle}
+    >
+      <LogHeader log={log} onClose={onClose} />
+
+      <div className="space-y-3 border-b-[1px] border-border py-4">
+        <div className="mt-[-24px]" />
+        <Headers headers={log.request_headers} title="Request Header" />
+        <Headers headers={flattenObject(JSON.parse(log.request_body))} title="Request Body" />
```

Comment:
this naming is weird, it's called "Headers" but then receives the request body oO

Reply:
When I first created that component it was only handling headers than evolved into something else 😄. Fair, I'll fix the name.